### PR TITLE
What to do when threads_per_worker is 0

### PR DESCRIPTION
**Pull request by jsignell**
*tl;dr Maybe `threads_per_worker` should throw an error if set to 0.*

While working on https://github.com/dask/distributed/pull/3984  I ran into some odd behaviors when `threads_per_worker` is set to 0 in `LocalCluster`. I am not sure what the intended behavior should be or if this PR makes things better. 

### Some examples:
On master the following results in a divide by 0 error. This PR sets n_workers to 0 which I think is a good solution:

```python
from distributed import LocalCluster, Client
cluster = LocalCluster(threads_per_worker=0)
client = Client(cluster)

client
# <Client: 'tcp://127.0.0.1:46121' processes=0 threads=0, memory=0 B> 
```

For the next case, I am not sure that the behavior should be. Here the `nthreads` is being passed int the `worker_kwargs` as 0. So we don't really have control over what happens after that. In this case it sets nthreads for each worker to `CPU_COUNT`

```python
from distributed import LocalCluster, Client

cluster = LocalCluster(n_workers=2, threads_per_worker=0)
client = Client(cluster)

client
# <Client: 'tcp://127.0.0.1:46509' processes=2 threads=24, memory=33.37 GB>
```

### CODE REVIEW

**Comment by jsignell** [comment]
@gforsyth who I was looking at this with.

**Comment by mrocklin** [comment]
I think that we should raise if `threads_per_worker < 1`

When testing in the distributed repository async tests are preferred, if they're easy to use.  You'll also want to use the `cleanup` fixture.

**Comment by jsignell** [comment]
> I think that we should raise if threads_per_worker < 1

That makes sense to me and should result in unambiguous behavior.

**Comment by jsignell** [comment]
@mrocklin I made it a warning instead of an error since it is easy enough to set it to None.

**Comment by mrocklin** [comment]
Thanks @jsignell !  Merged.
